postgres/datadog_checks/postgres/statements.py

```diff
@@ -0,0 +1,181 @@
+                ', '.join(list(missing_columns)),
+            )
+            return
+
+        desired_columns = (
+            list(PG_STAT_STATEMENTS_METRIC_COLUMNS.keys())
+            + list(PG_STAT_STATEMENTS_OPTIONAL_COLUMNS)
+            + list(PG_STAT_STATEMENTS_TAG_COLUMNS.keys())
+        )
+        query_columns = list(
+            set(desired_columns) & set(available_columns) + list(PG_STAT_STATEMENTS_TAG_COLUMNS.keys())
+        )
+        rows = self._execute_query(
+            db.cursor(cursor_factory=psycopg2.extras.DictCursor),
+            STATEMENTS_QUERY.format(
+                cols=', '.join(query_columns),
+                pg_stat_statements_view=self.config.pg_stat_statements_view,
+            ),
+            params=(self.config.dbname,),
+        )
+        if not rows:
+            return
+
+        def row_keyfunc(row):
+            # old versions of pg_stat_statements don't have a query ID so fall back to the query string itself
+            queryid = row['queryid'] if 'queryid' in row else row['query']
+            return (queryid, row['datname'], row['rolname'])
+
+        rows = self._state.compute_derivative_rows(rows, PG_STAT_STATEMENTS_METRIC_COLUMNS.keys(), key=row_keyfunc)
+        metric_limits = (
+            self.config.statement_metric_limits if self.config.statement_metric_limits else DEFAULT_METRIC_LIMITS
+        )
+        rows = apply_row_limits(rows, metric_limits, 'calls', True, key=row_keyfunc)
+
+        for row in rows:
+            try:
+                normalized_query = datadog_agent.obfuscate_sql(row['query'])
+                if not normalized_query:
+                    logger.warning("Query obfuscation resulted in empty query '%s'", row['query'])
+                    continue
+            except Exception as e:
+                logger.warning("Failed to obfuscate query '%s': %s", row['query'], e)
+                continue
+
+            # The APM resource hash will use the same query signature because the grouped query is close
+            # enough to the raw query that they will intersect frequently.
+            query_signature = compute_sql_signature(normalized_query)
+            tags = ['query_signature:' + query_signature, 'resource_hash:' + query_signature] + instance_tags
+            for column, tag_name in PG_STAT_STATEMENTS_TAG_COLUMNS.items():
+                if column not in row:
+                    continue
+                value = row[column]
+                if column == 'query':
+                    value = self._normalize_query_tag(value)
+                tags.append('{tag_name}:{value}'.format(tag_name=tag_name, value=value))
+
+            for column, metric_name in PG_STAT_STATEMENTS_METRIC_COLUMNS.items():
+                if column not in row:
+                    continue
+                value = row[column]
+                if column == 'total_time':
+                    # convert milliseconds to nanoseconds
+                    value = value * 1000000
```

Comment:
> I've read through #7837 (comment), but I'm not sure I understand why matching with the events/logs platform timestamp unit is important?

There is no hard requirement that EP events match DBM metrics. This is a standard chosen by our team and we will advocate for this standard for all database measures across all Datadog products (unless of course there is a strong reason not to use nanoseconds). We would prefer to work in integers over floats, and microseconds are too precise but milliseconds are not precise enough for duration metrics.

These metrics are extremely important to the DBM product and we wanted to standardize on a time unit across all databases to avoid bugs and confusion when we pull in data from all data sources (logs, integrations, rds, network, events, etc). 

> Do we not forward these values to the Agent, and then the Agent takes care of these conversions? Most likely I assume we're using a new Agent API instead, right?

I'm actually not aware of any sort of conversion like this, can you explain or point me in the right direction? From the metrics we're submitting now, the metrics API returns values which are in nanoseconds and the display applies the metric metadata to show the values in the appropriate unit. Given that the stored values on the backend are indeed in nanoseconds, I don't think the agent is doing any conversion.

> In any case, maybe let's consider writing a reusable helper (to indicate what conversion is being done), and then changing the comment to hint why we're making that conversion:

Yup, makes sense. Note that it will only be used in the Postgres integration though since Postgres stats are stored in milliseconds. MySQL performance_schema are stored in microseconds.

Reply:
> I don't think the agent is doing any conversion

Yes, you're right, thinking about it again I don't think there's any conversion done by the Agent either! So forget about that point of mine earlier. :-)

The rest sounds good to me, thanks a lot for these explanations!